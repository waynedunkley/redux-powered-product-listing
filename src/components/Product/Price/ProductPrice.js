import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import priceStyles from "./ProductPrice.style"

const StyledPrice = styled.p`
  ${priceStyles};
`

const propTypes = {
  price: PropTypes.string.isRequired,
}

const ProductPrice = ({ price }) => <StyledPrice>{price}</StyledPrice>

ProductPrice.propTypes = propTypes

export default ProductPrice
