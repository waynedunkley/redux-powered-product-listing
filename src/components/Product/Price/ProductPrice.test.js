import React from "react"
import { shallow } from "enzyme"
import ProductPrice from "./ProductPrice"

it("ProductPrice renders $3.99", () => {
  const price = "$3.99"
  const wrapper = shallow(<ProductPrice price={price} />)
  expect(wrapper.contains(price)).toEqual(true)
  expect(wrapper).toMatchSnapshot()
})
