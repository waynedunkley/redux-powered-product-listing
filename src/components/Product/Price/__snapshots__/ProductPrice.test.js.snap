// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductPrice renders $3.99 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductPrice
    price="$3.99"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "$3.99",
    },
    "ref": null,
    "rendered": "$3.99",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "$3.99",
      },
      "ref": null,
      "rendered": "$3.99",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
